documentation/src/main/antora/modules/ROOT/pages/component-checkpoint.adoc

```diff
@@ -0,0 +1,279 @@
+            log.info("[resume] No valid checkpoint configuration found, using start of dataset.");
+            bookmark = 0;
+        } else {
+            if ("finished".equals(configuration.checkpoint.status)) {
+                bookmark = 0;
+            } else {
+                bookmark = configuration.checkpoint.sinceId;
+            }
+        }
+        ...
+    }
+
+    @Producer
+    public JsonObject data() {
+        ...
+        final Integer produced = iterator.hasNext() ? iterator.next() : null;
+        if (produced == null) {
+            configuration.checkpoint.status = "finished";
+            configuration.checkpoint.sinceId = bookmark;
+            return null;
+        }
+        bookmark = produced;
+        configuration.checkpoint.sinceId = produced;
+        configuration.checkpoint.status = "running";
+
+        if (isCheckpointCondition(produced)) {
+            newBookmark = true;
+        }
+
+        return factory.createObjectBuilder().add("data", produced).build();
+    }
+
+    @CheckpointData
+    public Object getCheckpoint() {
+        newBookmark = false;
+        return configuration.checkpoint;
+    }
+
+    @CheckpointAvailable
+    public Boolean isCheckpointReady() {
+        return newBookmark;
+    }
+    ...
+----
+
+
+== Checkpointing Usage Scenarios
+At runtime, checkpointing can be used in two ways:
+
+=== 1. Explicit Runtime Usage
+In this mode, the runtime directly calls the methods of the input connector to manage checkpointing. It is responsible for:
+
+- Determining when to create a checkpoint.
+
+- Checking if a checkpoint is available using `isCheckpointReady()`.
+
+- Retrieving and storing the checkpoint using `getCheckpoint()`.
+
+Simple example of explicit checkpointing usage:
+[source,java]
+----
+  ...
+  input.start();
+  while ((input.next()) != null) {
+    if (input.isCheckpointReady()) {
+      serializeCheckpoint(input.getCheckpoint());
+    }
+  }
+  input.stop();
+----
+
+
+=== 2. Automatic Mode
+In this mode, checkpointing is handled automatically during the lifecycle:
+
+The `start()` method is called with a callback function that retrieves and serializes the checkpoint object.
+
+While reading records (`next()` method), the connector checks whether a checkpoint can be provided.
+
+If necessary, the checkpoint is generated by calling `getCheckpoint()`, which internally calls the method annotated with `@CheckpointData`.
+
+[source,java]
+----
+  final Consumer<CheckpointState> checkpointCallback = bookmark -> {
+    serializeCheckpoint(bookmark);
+  };
+  ...
+  input.start(checkpointCallback);
+  Record record;
+  while ((record = input.next()) != null) {
+    // process records
+  }
+  input.stop();
+----
+
+
+== How to resume
+
+As state previously, checkpoints are configured via the `@Checkpoint` annotated configuration class. The `@Checkpoint` annotation can be used to specify the method type used for checkpointing and the checkpointing frequency.
```

Comment:
IS that about to resume or checkpoint generation ?

Reply:
I'll say, it can be both...